### ISSUE TITLE
AttributeError due to incompatibility with newer OAuthlib version

### ISSUE THREAD

**Issue by easoncxz** [lepture/flask-oauthlib#238]
**TL;DR**: Flask-OAuthlib is out-of-date from OAuthlib and is hence causing exceptions to be raised. (I don't know how to fix it.)

Currently I'm seeing that `master` points to commit [cf24784fe4](https://github.com/lepture/flask-oauthlib/tree/cf24784fe47e434edf6e22f71b95251ffdc921da).

As I was using Flask-OAuthlib, [line 822 of flask_oauthlib/provider/oauth1.py](https://github.com/lepture/flask-oauthlib/blob/cf24784fe47e434edf6e22f71b95251ffdc921da/flask_oauthlib/provider/oauth1.py#L822) throws `AttributeError: request_token`. Here's my own stack trace at some point (it's not actually important; just posting for completeness):

    Traceback (most recent call last):
      File "/Users/eason/Documents/2015S2/SE700B/flask_oauthlib_provider/.venv/lib/python3.4/site-packages/flask/app.py", line 1836, in __call__
        return self.wsgi_app(environ, start_response)
      File "/Users/eason/Documents/2015S2/SE700B/flask_oauthlib_provider/.venv/lib/python3.4/site-packages/flask/app.py", line 1820, in wsgi_app
        response = self.make_response(self.handle_exception(e))
      File "/Users/eason/Documents/2015S2/SE700B/flask_oauthlib_provider/.venv/lib/python3.4/site-packages/flask/app.py", line 1403, in handle_exception
        reraise(exc_type, exc_value, tb)
      File "/Users/eason/Documents/2015S2/SE700B/flask_oauthlib_provider/.venv/lib/python3.4/site-packages/flask/_compat.py", line 33, in reraise
        raise value
      File "/Users/eason/Documents/2015S2/SE700B/flask_oauthlib_provider/.venv/lib/python3.4/site-packages/flask/app.py", line 1817, in wsgi_app
        response = self.full_dispatch_request()
      File "/Users/eason/Documents/2015S2/SE700B/flask_oauthlib_provider/.venv/lib/python3.4/site-packages/flask/app.py", line 1477, in full_dispatch_request
        rv = self.handle_user_exception(e)
      File "/Users/eason/Documents/2015S2/SE700B/flask_oauthlib_provider/.venv/lib/python3.4/site-packages/flask/app.py", line 1381, in handle_user_exception
        reraise(exc_type, exc_value, tb)
      File "/Users/eason/Documents/2015S2/SE700B/flask_oauthlib_provider/.venv/lib/python3.4/site-packages/flask/_compat.py", line 33, in reraise
        raise value
      File "/Users/eason/Documents/2015S2/SE700B/flask_oauthlib_provider/.venv/lib/python3.4/site-packages/flask/app.py", line 1475, in full_dispatch_request
        rv = self.dispatch_request()
      File "/Users/eason/Documents/2015S2/SE700B/flask_oauthlib_provider/.venv/lib/python3.4/site-packages/flask/app.py", line 1461, in dispatch_request
        return self.view_functions[rule.endpoint](**req.view_args)
      File "/Users/eason/Documents/2015S2/SE700B/flask_oauthlib_provider/.venv/lib/python3.4/site-packages/flask_oauthlib/provider/oauth1.py", line 402, in decorated
        uri, http_method=http_method, body=body, headers=headers
      File "/Users/eason/Documents/2015S2/SE700B/flask_oauthlib_provider/.venv/lib/python3.4/site-packages/oauthlib/oauth1/rfc5849/endpoints/authorization.py", line 156, in get_realms_and_credentials
        request.resource_owner_key, request):
      File "/Users/eason/Documents/2015S2/SE700B/flask_oauthlib_provider/.venv/lib/python3.4/site-packages/flask_oauthlib/provider/oauth1.py", line 822, in verify_request_token
        tok = request.request_token or self._grantgetter(token=token)
      File "/Users/eason/Documents/2015S2/SE700B/flask_oauthlib_provider/.venv/lib/python3.4/site-packages/oauthlib/common.py", line 412, in __getattr__
        raise AttributeError(name)
    AttributeError: request_token

I found Travis job [#341.4](https://travis-ci.org/lepture/flask-oauthlib/jobs/72856247) to have a stack trace that's very similar to mine --- even line numbers are the same. Travis job [#345.4](https://travis-ci.org/lepture/flask-oauthlib/jobs/73519005), which is a few commits away from 341.4, does not show this AttributeError. Doing a manual diff reveals that the author ( @therealmarv ) changed `requirements.txt` between those two builds. You can download snapshots of those two commits and do a manual diff, but here's the interesting segment of it for convenience:

    --- flask-oauthlib-fbeae3e5a1f33b7f68ba253bb01fcc8bca4f712a/requirements.txt	Tue Jul 28 03:40:40 2015
    +++ flask-oauthlib-44bdfe4331afa86bd3613768a05bf7b05b9451f2/requirements.txt	Fri Jul 31 20:46:35 2015
    @@ -1,5 +1,5 @@
    +Flask==0.10.1
    +mock==1.3.0
    +oauthlib==0.7.2
    +requests-oauthlib==0.5.0
    +Flask-SQLAlchemy==2.0
    -Flask
    -Mock
    -oauthlib
    -requests-oauthlib
    -Flask-SQLAlchemy

I noticed `requirements.txt` in my own project to say

    oauthlib==1.0.3

whereas Flask-OAuthlib's says

    oauthlib==0.7.2

There is indeed a "Breaking Change" as per the [1.0.0](https://github.com/idan/oauthlib/blob/master/CHANGELOG.rst#100-2015-07-19) entry in OAuthlib's changelog that started raising attribute errors.

To reproduce some stack trace like mine that is caused by the same AttributeError, do something like the below:

    git clone git@github.com:lepture/flask-oauthlib.git
    git checkout master
    # virtualenv and activate
    pip install -r requirements.txt
    pip install -U oauthlib    # currently on version 1.0.3 as noted above
    # run Nose tests and see them fail with stack traces caused by AttributeErrors.

**Comment by lepture** [comment]
report it here: https://github.com/idan/oauthlib/blob/master/oauthlib/common.py

**Comment by lepture** [comment]
Send PR to OAuthlib.
